From cb0c19d3c37426fb3e3455b50517e457ca682037 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sat, 19 Jul 2014 04:28:17 -0400
Subject: [pci] Remove unused alloc_pci_dev()

Message-id: <20140719042817.19438.33225.stgit@gir.stowe>
Patchwork-id: 86015
O-Subject: [RHEL7.1 PATCH 53/81] PCI: Remove unused alloc_pci_dev()
Bugzilla: 1110895
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110895
Upstream Status: e2760c54a4f5131867bf1b35d59169267d35d3d0

commit e2760c54a4f5131867bf1b35d59169267d35d3d0
Author: Stephen Hemminger <stephen@networkplumber.org>
Date:   Fri Jan 10 15:46:34 2014 -0700

    PCI: Remove unused alloc_pci_dev()

    My philosophy is unused code is dead code.  And dead code is subject to bit
    rot and is a likely source of bugs.  Use it or lose it.

    This removes this unused and deprecated interface:

        alloc_pci_dev()

    [bhelgaas: split to separate patch]
    Signed-off-by: Stephen Hemminger <stephen@networkplumber.org>
    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/pci/probe.c b/drivers/pci/probe.c
index 0b5a010..7f4a52c 100644
--- a/drivers/pci/probe.c
+++ b/drivers/pci/probe.c
@@ -1309,12 +1309,6 @@ struct pci_dev *pci_alloc_dev(struct pci_bus *bus)
 }
 EXPORT_SYMBOL(pci_alloc_dev);
 
-struct pci_dev *alloc_pci_dev(void)
-{
- return pci_alloc_dev(NULL);
-}
-EXPORT_SYMBOL(alloc_pci_dev);
-
 bool pci_bus_read_dev_vendor_id(struct pci_bus *bus, int devfn, u32 *l,
      int crs_timeout)
 {
diff --git a/include/linux/pci.h b/include/linux/pci.h
index cb559f0..837e257 100644
--- a/include/linux/pci.h
+++ b/include/linux/pci.h
@@ -393,7 +393,6 @@ static inline struct pci_dev *pci_physfn(struct pci_dev *dev)
 }
 
 struct pci_dev *pci_alloc_dev(struct pci_bus *bus);
-struct pci_dev * __deprecated alloc_pci_dev(void);
 
 #define to_pci_dev(n) container_of(n, struct pci_dev, dev)
 #define for_each_pci_dev(d) while ((d = pci_get_device(PCI_ANY_ID, PCI_ANY_ID, d)) != NULL)
-- 
1.7.1